### PR TITLE
Update cognite sdk

### PR DESCRIPTION
**Pull request by doctrino**
Biggest change is that all container properties are now listable, so there are many extra checks that are no longer necessary.

Updating lead to some other changes as well.

### CODE REVIEW

File: cognite/neat/rules/models/_rules/domain_rules.py [hunk 1]
```diff
@@ -24,6 +25,18 @@ class DomainProperty(SheetEntity):
     min_count: int | None = Field(alias="Min Count", default=None)
     max_count: int | float | None = Field(alias="Max Count", default=None)
 
+    @field_serializer("max_count", when_used="json-unless-none")
+    def serialize_max_count(self, value: int | float | None) -> int | float | None | str:
+        if isinstance(value, float) and math.isinf(value):
+            return None
```
Comment (doctrino) [inline cognite/neat/rules/models/_rules/domain_rules.py:31]: @nikokaoja I think this should solve the issue with json inf.

File: pyproject.toml
Comment (nikokaoja) [inline pyproject.toml:1]: Bump version to 0.73 ? so @CimStordal can pull latest neat to fix issues he has in `docparser`

File: cognite/neat/rules/exporter/_rules2dms.py [hunk 2]
```diff
@@ -284,12 +283,9 @@ def containers_from_rules(cls, rules: Rules) -> dict[str, ContainerApply]:
                     if (
                         not isinstance(existing_property.type, DirectRelation)
                         and not isinstance(api_container_property.type, DirectRelation)
-                        and cast(ListablePropertyType, existing_property.type).is_list
-                        != cast(ListablePropertyType, api_container_property.type).is_list
+                        and existing_property.type.is_list != api_container_property.type.is_list
```
Comment (nikokaoja) [inline cognite/neat/rules/exporter/_rules2dms.py:286]: great !

File: cognite/neat/rules/models/_rules/information_rules.py [hunk 3]
```diff
@@ -69,7 +71,7 @@
 class InformationMetadata(BaseMetadata):
     role: ClassVar[RoleTypes] = RoleTypes.information_architect
     schema_: SchemaCompleteness = Field(alias="schema")
-    extension: ExtensionCategory | None = ExtensionCategory.addition
```
Comment (nikokaoja) [inline cognite/neat/rules/models/_rules/information_rules.py:72]: it was not serializing correctly ?

File: cognite/neat/rules/models/_rules/information_rules.py [hunk 4]
```diff
@@ -156,6 +158,18 @@ class InformationProperty(SheetEntity):
     )
     comment: str | None = Field(alias="Comment", default=None)
 
+    @field_serializer("max_count", when_used="json-unless-none")
```
Comment (nikokaoja) [inline cognite/neat/rules/models/_rules/information_rules.py:161]: great !

**Review by nikokaoja — APPROVED** [review]

File: cognite/neat/rules/models/_rules/information_rules.py (hunk 3 above)
Comment (doctrino) [reply cognite/neat/rules/models/_rules/information_rules.py:72]: I updated pydantic and then it stopped serializing as expected.

**Pull request merged**